### PR TITLE
Icons: Rewrite using lib/fs.js

### PR DESCRIPTION
**Pull request by DeathlyPlays**

### CODE REVIEW

File: wavelength-plugins/icons.js [hunk 1]
```diff
@@ -1,84 +1,87 @@
...
-let icons = {};
-const fs = require('fs');
+const FS = require("../lib/fs.js");
 
-function load() {
-	fs.readFile('config/icons.json', 'utf8', function (err, file) {
-		if (err) return;
-		icons = JSON.parse(file);
-	});
+let icons = FS("config/icons.json").readIfExistsSync();
```
Comment (Mystifi) [inline wavelength-plugins/icons.js]: This should make it easier for TypeScripting in the future:
```js
let iconsData = FS("config/icons.json").readIfExistsSync();
let icons = {};

if (iconsData) {
   icons = JSON.parse(icons);
}
```

File: wavelength-plugins/icons.js [hunk 2]
```diff
@@ -1,84 +1,87 @@
...
+	let css = "";
 	name = toId(name);
-	Rooms.rooms.forEach((curRoom, id) => {
-		if (curRoom.id === 'global' || curRoom.type !== 'chat' || curRoom.isPersonal) return;
-		if (!isNaN(Number(id.charAt(0)))) return;
-		rooms.push('#' + id + '-userlist-user-' + name);
-	});
-	css = rooms.join(', ');
-	css += '{\nbackground: url("' + icon + '") no-repeat right\n}\n';
+	css = `[id$="-userlist-user-${name}] {\nbackground: url("${icon}") no-repeat right !important;\n}\n`;
```
Comment (Mystifi) [inline wavelength-plugins/icons.js]: Thank *God* for changing this. Oh my god. The amount of CSS reduced...you're doing God's work, kiddo.

Comment (DeathlyPlays) [reply wavelength-plugins/icons.js]: Thanks LOL

**Comment by DeathlyPlays** [comment]
Done @Mystifi :P

File: wavelength-plugins/icons.js [hunk 3]
```diff
@@ -1,84 +1,85 @@
...
 		newCss += generateCSS(name, icons[name]);
 	}
-	newCss += '/* ICONS END */\n';
+	newCss += "/* ICONS END */\n";
 
-	let file = fs.readFileSync('config/custom.css', 'utf8').split('\n');
-	if (~file.indexOf('/* ICONS START */')) file.splice(file.indexOf('/* ICONS START */'), (file.indexOf('/* ICONS END */') - file.indexOf('/* ICONS START */')) + 1);
-	fs.writeFileSync('config/custom.css', file.join('\n') + newCss);
+	let file = FS("config/custom.css").readIfExistsSync().split("\n");
+	if (!file.includes("/* ICONS START */")) file.splice(file.indexOf("/* ICONS START */"), (file.indexOf("/* ICONS END */") - file.indexOf("/* ICONS START */")) + 1);
```
Comment (Mystifi) [inline wavelength-plugins/icons.js]: I'm sorry; remove the `!`. I was being dumb.

Comment (DeathlyPlays) [reply wavelength-plugins/icons.js]: ok

**Comment by DeathlyPlays** [comment]
@HoeenCoder @Mystifi bump

**Comment by Volco** [comment]
@DeathlyPlays this is 100% tested, no issues?

**Comment by DeathlyPlays** [comment]
Yes @Volco

File: wavelength-plugins/icons.js [hunk 4]
```diff
@@ -1,84 +1,87 @@
...
-			this.sendReply("You removed " + target + "'s icon.");
-			Monitor.adminlog(user.name + " removed " + target + "'s icon.");
-			this.privateModAction("(" + target + "'s icon was removed by " + user.name + ".)");
-			this.modlog('ICON', target, `Removed icon`);
-			if (Users(target) && Users(target).connected) Users(target).popup("|html|" + WL.nameColor(user.name, true) + " has removed your userlist icon.");
+			this.sendReply(`You removed ${target}'s icon.`);
+			Monitor.log(`${user.name} removed ${target}'s icon.`);
+			this.privateModAction(`(${target}'s icon was removed by ${user.name}.)`);
+			this.modlog("ICON", target, `Removed icon`);
+			if (Users(target) && Users(target).connected) Users(target).popup(`|html|${WL.nameColor(user.name, true)} has removed your userlist icon.`);
```
Comment (Volco) [inline wavelength-plugins/icons.js]: ``if (Users(toId(target)) && Users(toId(target)).connected)``

File: wavelength-plugins/icons.js [hunk 5]
```diff
@@ -1,84 +1,87 @@
...
-			Monitor.adminlog(target[0] + " has received an icon from " + user.name + ".");
-			this.privateModAction("|raw|(" + target[0] + " has recieved icon: <img src='" + target[1] + "' width='32' height='32'> from " + user.name + ".)");
-			this.modlog('ICON', target[0], `Set icon to ${target[1]}`);
-			if (Users(target[0]) && Users(target[0]).connected) Users(target[0]).popup("|html|" + WL.nameColor(user.name, true) + " has set your userlist icon to: <img src='" + target[1] + "' width='32' height='32'><br><center>Refresh, If you don't see it.</center>");
+			this.sendReply(`|raw|You have given ${WL.nameColor(target[0], true)} an icon.`);
+			Monitor.log(`${target[0]} has received an icon from ${user.name}.`);
+			this.privateModAction(`|raw|(${target[0]} has received icon: <img src="${target[1]}" width="32" height="32"> from ${user.name}.)`);
+			this.modlog("ICON", target[0], `Set icon to ${target[1]}`);
+			if (Users(target[0]) && Users(target[0]).connected) Users(target[0]).popup(`|html|${WL.nameColor(user.name, true)} has set your userlist icon to: <img src="${target[1]}" width="32" height="32"><br /><center>Refresh, If you don't see it.</center>`);
```
Comment (Volco) [inline wavelength-plugins/icons.js]: ``if (Users(toId(target[0])) && Users(toId(target[0])).connected)``

**Comment by Volco** [comment]
@DeathlyPlays do the changes i've requested please

**Comment by DeathlyPlays** [comment]
You do realize my Dad died @Volco..... Kinda got a life to deal with

**Comment by Volco** [comment]
>implying you told me when you didnt. 
and sorry to hear that but next time before being a jerk about it maybe tell the person first?

**Pull request merged**